File: app/src/main/java/com/example/sony/newsapi/screen/newslist/NewsListContract.java
package com.example.sony.newsapi.screen.newslist;

import com.example.sony.newsapi.model.Article;

import java.util.List;

public interface NewsListContract {

    interface View {
        void showTitle();
        void hideSwipeRefreshLayout();
        void showNewsList(List<Article> newsList);
        void showErrorToast();
    }

    interface Presenter {
        void loadNewsList();
    }
}

File: app/src/main/java/com/example/sony/newsapi/screen/newslist/NewsListContract.kt
package com.example.sony.newsapi.screen.newslist

import com.example.sony.newsapi.model.Article

interface NewsListContract {

    interface View {
        fun showTitle()
        fun hideSwipeRefreshLayout()
        fun showNewsList(newsList: List<Article>)
        fun showErrorToast()
    }

    interface Presenter {
        fun loadNewsList()
    }
}
